### PR TITLE
[gestalt]: Update gestalt definition to be in sync with 14.26

### PR DESCRIPTION
**Pull request by czgu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://gestalt.netlify.app/What's%20New
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

P.S.
Removed all refs definitions since it causes errors when passing props from a parent component to a gestalt props

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## gestalt ([<kbd>unpkg</kbd>](https://unpkg.com/browse/gestalt@latest/))
was missing the following properties:
1. Collection
1. CompositeZIndex
1. FixedZIndex
1. MasonryDefaultLayout
1. MasonryUniformRowLayout

<details>
<summary>as well as these 4 other properties...</summary>
<p>ScrollFetch, useColorScheme, useFocusVisible, useReducedMotion</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a8bb212b124c4062457be0b38a1e362d66d892c9
</p>

File: types/gestalt/index.d.ts [hunk 1]
```diff
@@ -718,11 +756,29 @@ export interface ModalProps {
     closeOnOutsideClick?: boolean;
     footer?: React.ReactNode;
     heading?: string | React.ReactNode;
-    ref?: React.Ref<'div'>;
     role?: 'alertdialog' | 'dialog';
     size?: 'sm' | 'md' | 'lg' | number;
 }
 
+/*
```
Comment (k-yle) [inline types/gestalt/index.d.ts:763]: just FYI, these comments won't be shown in intellisense unless they're jsdoc comments, i.e. they start with `/**` not `/*`

Comment (czgu) [reply types/gestalt/index.d.ts]: thanks, added jsdoc formatting for all props

File: types/gestalt/index.d.ts [hunk 2]
```diff
@@ -197,35 +215,39 @@ export interface ButtonProps {
         | React.KeyboardEvent<HTMLAnchorElement>
         | React.KeyboardEvent<HTMLButtonElement>
     >;
-    ref?: React.Ref<'button'> | React.Ref<'a'>;
```
Comment (k-yle) [inline types/gestalt/index.d.ts:200]: how does this affect https://gestalt.netlify.app/Button#ref ? Seems like passing `ref` to button is a documented use-case

File: types/gestalt/index.d.ts [hunk 3]
```diff
@@ -161,24 +183,20 @@ export interface BoxProps {
     mdPaddingY?: UnsignedUpTo12;
     lgPaddingY?: UnsignedUpTo12;
     position?: 'static' | 'absolute' | 'relative' | 'fixed';
-    ref?: React.Ref<'div'>;
```
Comment (k-yle) [inline types/gestalt/index.d.ts:164]: as per my other comment, why did you remove the `ref` prop from these components? My understanding is that it's needed to make `<Flyout />` show the popup on top of the right element - see https://gestalt.netlify.app/Flyout#basicExample

**Review by k-yle — CHANGES_REQUESTED** [review]
See my comments - although I may have misunderstood how `ref`s work with `<Flyout />`

**Comment by czgu** [comment]
> See my comments - although I may have misunderstood how `ref`s work with `<Flyout />`

Oh the old typing made forwarding props very hard, for example
```typescript
type BoxWithTextProps = { text: string } & BoxProps;
const BoxWithText: React.FC<BoxWithTextProps> = ({ text, ...otherProps }) => <Box {...otherProps}>{text}</Box>
```
would give an error about wrong ref types.

I just changed the class export into const export with React ForwardRef which should fix this issue

**Review by k-yle — APPROVED** [review]
cool, thanks for all this work! 👍

**Comment by k-yle** [comment]
Ready to merge
